tests/test_gitlab.py

```diff
@@ -129,6 +130,43 @@ def test_to_taskwarrior(self):
             self.arbitrary_extra
         )
 
+        expected_output = {
+            'project': self.arbitrary_extra['project'],
+            'priority': self.service.default_priority,
+            'annotations': [],
+            'tags': [u'feature'],
+            issue.URL: self.arbitrary_extra['issue_url'],
+            issue.REPO: 'project',
+            issue.STATE: self.arbitrary_issue['state'],
+            issue.TYPE: self.arbitrary_extra['type'],
+            issue.TITLE: self.arbitrary_issue['title'],
+            issue.NUMBER: self.arbitrary_issue['iid'],
+            issue.UPDATED_AT: self.arbitrary_updated.replace(microsecond=0),
+            issue.CREATED_AT: self.arbitrary_created.replace(microsecond=0),
+            issue.DUEDATE: self.arbitrary_duedate,
+            issue.DESCRIPTION: self.arbitrary_issue['description'],
+            issue.MILESTONE: self.arbitrary_issue['milestone']['title'],
+            issue.UPVOTES: 0,
+            issue.DOWNVOTES: 0,
+            issue.WORK_IN_PROGRESS: 1,
+            issue.AUTHOR: 'john_smith',
+            issue.ASSIGNEE: 'jack_smith',
+            issue.NAMESPACE: 'arbitrary_namespace',
+            issue.WEIGHT: 3,
+        }
+        actual_output = issue.to_taskwarrior()
+
+        self.assertEqual(actual_output, expected_output)
+
+    def test_work_in_progress(self):
+        arbitrary_issue_2 = self.arbitrary_issue
```

Comment:
Assignment like this just adds another label for `self.arbitrary_issue`, it does not copy it.

You could copy it with `arbitrary_issue_2 = copy.deepcopy(self.arbitrary_issue)`, but it would probably be better to skip the copying/relabeling altogether and move the `arbitrary_*` variables from the class scope to the `setUp` method to guarantee testcase isolation.

Reply:
I understand the problem here...

Do you suggest to define all ```arbitrary_*``` members in the ```setUp``` method instead directily in the class definition? So they would still be members, but defined in the ```setUp``` method which will be called before each test function, is this correct? (Sorry, I am quite new to python unittests).

So I would define ```arbitrary_issue``` as before, but inside the ```setUp``` function and then create ```arbitrary_issue_no_wip``` by creating a deepcopy from ```arbitrary_issue``` and modifying the respective field. I would still do it inside the test function, as this is only relevant for that scope, right?